### PR TITLE
Query multiple tables in animals table vet clinic sql

### PR DESCRIPTION
**Pull request by richie1988**
### Created a table named owners with the following columns:

- id: integer (set it as autoincremented PRIMARY KEY)
- full_name: string
- age: integer
- Create a table named species with the following columns:
- id: integer (set it as autoincremented PRIMARY KEY)
- name: string
- Modify animals table:
- Make sure that id is set as autoincremented PRIMARY KEY
- Removed column species
- Added column species_id which is a foreign key referencing species table
- Added column owner_id which is a foreign key referencing the owners table
 ### I Inserted the following data into the owners table:
- Sam Smith 34 years old.
- Jennifer Orwell 19 years old.
- Bob 45 years old.
- Melody Pond 77 years old.
- Dean Winchester 14 years old.
- Jodie Whittaker 38 years old.

### I Inserted the following data into the species table:

- Pokemon
- Digimon
- I Modified my animals so it includes the species_id value:
- If the name ends in "mon" it will be Digimon
- All other animals are Pokemon
- Modified My Animals table to include owner information (owner_id):
- Sam Smith owns Agumon.
- Jennifer Orwell owns Gabumon and Pikachu.
- Bob owns Devimon and Plantmon.
- Melody Pond owns Charmander, Squirtle, and Blossom.
- Dean Winchester owns Angemon and Boarmon.

### I Wrote queries

- What animals belong to Melody Pond?
- List of all animals that are pokemon (their type is Pokemon).
- List all owners and their animals, remember to include those that don't own any animal.
- How many animals are there per species?
- List all Digimon owned by Jennifer Orwell.
- List all animals owned by Dean Winchester that haven't tried to escape.
- Who owns the most animals?
Code_tTo1XdUwvP
Code_uz945LS9DJ
Code_wIRtcT69s0
Code_1c98Wiacqk
Code_fKcCsigxqk
Code_ISd8UB3hYZ
Code_JwJzD0nXar
Code_qy011IvIeX
Code_J6NcBKJjJq

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -14,3 +14,77 @@ INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_
...
     ('Ditto', '2022-05-14', 4, TRUE, 22.00, 'unspecified');
+
+    
+/* day 3 of the query multiple tables*/
+-- Create the owners table
+CREATE TABLE owners (
+    id INT AUTO_INCREMENT PRIMARY KEY,
+    full_name VARCHAR(255),
+    age INT
+);
```
Comment (shahadat3669) [inline data.sql]: 👍 Good progress on adding the indexes! However, there's a small adjustment to make to align with the curriculum guidelines.

- [ ] Currently, your `owners ` creation operation is inside the `datas.sql` file. According to the [curriculum-databases-projects-template](https://github.com/microverseinc/curriculum-template-databases#:~:text=This%20repository%20includes,Use%20them%20responsibly!), `CREATE` operations should be inside the `schema.sql` file. Moreover, you already have the same operation inside the `schema.sql` file. I will recommend you remove this operation to align with the requirement.

By organizing your SQL scripts this way, you'll ensure that your project structure aligns with the curriculum guidelines. Keep up the great work! 👍📜

File: data.sql [hunk 2]
```diff
@@ -14,3 +14,77 @@ INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_
...
+    ('Bob', 45),
+    ('Melody Pond', 77),
+    ('Dean Winchester', 14),
+    ('Jodie Whittaker', 38);
+
+-- Create the species table
+CREATE TABLE species (
+    id INT AUTO_INCREMENT PRIMARY KEY,
+    name VARCHAR(255)
+);
```
Comment (shahadat3669) [inline data.sql]: 👍 Good progress on adding the indexes! However, there's a small adjustment to make to align with the curriculum guidelines.

- [ ] Currently, your `species ` creation operation is inside the `datas.sql` file. According to the [curriculum-databases-projects-template](https://github.com/microverseinc/curriculum-template-databases#:~:text=This%20repository%20includes,Use%20them%20responsibly!), `CREATE` operations should be inside the `schema.sql` file. Moreover, you already have the same operation inside the `schema.sql` file. I will recommend you remove this operation to align with the requirement.

By organizing your SQL scripts this way, you'll ensure that your project structure aligns with the curriculum guidelines. Keep up the great work! 👍📜

File: data.sql [hunk 3]
```diff
@@ -14,3 +14,77 @@ INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_
...
+-- Insert data into the species table
+INSERT INTO species (name) VALUES
+    ('Pokemon'),
+    ('Digimon');
+
+ALTER TABLE animals
+MODIFY COLUMN id INT AUTO_INCREMENT PRIMARY KEY,
+DROP COLUMN species,
+ADD COLUMN species_id INT,
+ADD COLUMN owner_id INT;
```
Comment (shahadat3669) [inline data.sql]: 👍 Good progress on adding the indexes! However, there's a small adjustment to make to align with the curriculum guidelines.

- [ ] Currently, your `alter` operation is inside the `datas.sql` file. According to the [curriculum-databases-projects-template](https://github.com/microverseinc/curriculum-template-databases#:~:text=This%20repository%20includes,Use%20them%20responsibly!), `CREATE` operations like `alter` should be inside the `schema.sql` file. Moreover, you already have the same operation inside the `schema.sql` file. I will recommend you remove this operation to align with the requirement.

By organizing your SQL scripts this way, you'll ensure that your project structure aligns with the curriculum guidelines. Keep up the great work! 👍📜

File: queries.sql [hunk 4]
```diff
@@ -77,3 +77,45 @@ SELECT species, AVG(escape_attempts) AS avg_escape_attempts
 FROM animals
 WHERE YEAR(date_of_birth) BETWEEN 1990 AND 2000
 GROUP BY species;
+
+/* Day 3 of query multiple tables*/
+
+SELECT a.name
+FROM animals AS a
+JOIN owners AS o ON a.owner_id = o.id
+WHERE o.full_name = 'Melody Pond';
```
Comment (shahadat3669) [inline queries.sql:86]: 👍 Great work on your query! However, there's one important step remaining:

- [ ] As per the project requirement, please update the screenshot for this query. Your screenshot should reflect this query, showing the results. 


By updating the screenshot to align with the project requirements, your documentation will be complete and accurate. Keep up the great work! 👍📊🔄

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi @richie1988 , 

Great job so far! 👏 You've done well, and you're almost there for the next project ⏭️, but there are a few issues 🐛 that you still need to work on.




## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Highlights
✅ Everything that is a query that helps to analyze data are saved in the queries.sql file.
✅ add a new table with the name `owners`
✅ add a new table with the name `species`


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadata3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 2 above)
Comment (richie1988) [reply data.sql]: Thank you for your review i will look into that

**Comment by richie1988** [comment]
**As requested by the previoues reviewer this is the screenshots and check the updated i made **

Code_J6NcBKJjJq

File: queries.sql (hunk 4 above)
Comment (danmainah) [reply queries.sql:86]: - [ ] @richie1988 Great work implementing the changes requested, however, the change should be implemented in the pull request description, kindly consider updating the screenshot as previously requested.
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @richie1988 , 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️ Implemented the change requested to ensure all CREATE operations are inside the schema.sql file and the data.sql only handles adding data to the database.

 # Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by richie1988** [comment]
### This is the screen shot i did add 
Code_J6NcBKJjJq

**Review by frank1738 — APPROVED** [review]
Hi @richie1988 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights


✅ Descriptive PR Description
✅ Great work updating the screenshot results on your PR description👍 


## Optional suggestions


#### 👍 Great work on your PR! Just a couple of friendly suggestions:

- [ ] To keep your readme spick and span 🧹, consider trimming down the descriptive text. For instance, you can remove phrases like "Describe the tech stack" and include only the sections relevant to your project. Remember, they're just here to guide you, not to stay in the spotlight.

- [ ] It seems like the MIT link 🚧 is broken. A quick fix would be greatly appreciated!

#### Thanks for your hard work! 👏

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**